From f7a6816242d4357fbfa10f2752b297cee118f88d Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Sun, 4 Apr 2021 12:24:14 +0200
Subject: [PATCH] Upgrade websocket-client to 0.58.0

Seems that since websocket-client 0.54.0 it has started passing the
WebSocketApp object itself to the callbacks. I don't see this happening
in the source itself, seems nothing has changed there, but accepting
that extra argument does fix the problem with a newer websocket-client
---
 mycroft/messagebus/client/client.py | 6 +++---
 requirements/requirements.txt       | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/mycroft/messagebus/client/client.py b/mycroft/messagebus/client/client.py
index 68aec51fb73..9bba22d1311 100644
--- a/mycroft/messagebus/client/client.py
+++ b/mycroft/messagebus/client/client.py
@@ -110,7 +110,7 @@ class MessageBusClient:
             on_message=self.on_message
         )
 
-    def on_open(self):
+    def on_open(self, _):
         LOG.info("Connected")
         self.connected_event.set()
         self.emitter.emit("open")
@@ -120,7 +120,7 @@ class MessageBusClient:
     def on_close(self):
         self.emitter.emit("close")
 
-    def on_error(self, error):
+    def on_error(self, _, error):
         """ On error start trying to reconnect to the websocket. """
         if isinstance(error, WebSocketConnectionClosedException):
             LOG.warning('Could not send message because connection has closed')
@@ -146,7 +146,7 @@ class MessageBusClient:
         except WebSocketException:
             pass
 
-    def on_message(self, message):
+    def on_message(self, _, message):
         parsed_message = Message.deserialize(message)
         self.emitter.emit('message', message)
         self.emitter.emit(parsed_message.msg_type, parsed_message)
diff --git a/requirements/requirements.txt b/requirements/requirements.txt
index 8c22b4a4c5a..d4d512084d1 100644
--- a/requirements/requirements.txt
+++ b/requirements/requirements.txt
@@ -4,7 +4,7 @@ PyAudio==0.2.11
 pyee==8.1.0
 SpeechRecognition==3.8.1
 tornado==6.0.3
-websocket-client==0.54.0
+websocket-client==0.58.0
 requests-futures==0.9.5
 pyserial==3.0
 psutil==5.6.6
-- 
2.31.1

